services/src/main/java/org/keycloak/authentication/authenticators/conditional/ConditionalRoleAuthenticatorFactory.java

```diff
@@ -81,7 +73,19 @@ public String getHelpText() {
 
     @Override
     public List<ProviderConfigProperty> getConfigProperties() {
-        return commonConfig;
+        ProviderConfigProperty role = new ProviderConfigProperty();
+        role.setType(ProviderConfigProperty.ROLE_TYPE);
+        role.setName(CONDITIONAL_USER_ROLE);
+        role.setLabel("User role");
+        role.setHelpText("Role the user should have to execute this flow. Click 'Select Role' button to browse roles, or just type it in the textbox. To specify an application role the syntax is appname.approle, i.e. myapp.myrole");
+
+        ProviderConfigProperty negateOutput = new ProviderConfigProperty();
+        negateOutput.setType(ProviderConfigProperty.BOOLEAN_TYPE);
+        negateOutput.setName(CONF_NEGATE);
+        negateOutput.setLabel("Negate output");
+        negateOutput.setHelpText("Apply a not to the check result");
```

Comment:
My vote would be to add more details to this help text to make it more obvious for the administrator. Like "Apply a not to the check result. When this is true, then the condition will evaluate to true just if user does NOT have the specified role. When this is false, the condition will evaluate to true just if user has the specified role".